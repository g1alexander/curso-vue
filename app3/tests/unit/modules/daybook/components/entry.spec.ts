import { shallowMount } from "@vue/test-utils";
import EntryVue from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mocks/journal-state";

describe("tests componet Entry", () => {
  const $router = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(EntryVue, {
    props: {
      entry: journalState.entries[0],
    },
    global: {
      mocks: {
        $router,
      },
    },
  });

  test("debe de hacer to match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de redireccionar al hacer click en el entry-container", () => {
    const div = wrapper.find(".entry-container");
    div.trigger("click");

    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: journalState.entries[0].id },
    });
  });

  test("pruebas en las propiedades computadas", () => {
    const date = journalState.entries[0].date;
    // console.log(date); //Tue Mar 22 2022
    expect(wrapper.vm.day).toBe(22);
    expect(wrapper.vm.mouth).toBe("Marzo");
    expect(wrapper.vm.yearDay).toBe("2022, Martes");
  });
});
