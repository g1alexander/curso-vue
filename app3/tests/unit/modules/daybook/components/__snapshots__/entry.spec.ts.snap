// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests componet Entry debe de hacer to match snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- title -->
  <div class="entry-title d-flex"><span class="text-success fd-5 fw-bold">22</span><span class="mx-1 fs-5">Marzo</span><span class="mx-2 fw-light">2022, Martes</span></div>
  <div class="entry-description">Entrada numero 1</div>
</div>
`;
